solutions/ceoi-19-AmusementPark.mdx

```diff
@@ -0,0 +1,112 @@
+---
+id: ceoi-19-AmusementPark
+source: CEOI 2019
+title: 2019 - Amusement Park
+author: Shijie Ren
+---
+
+[CF Editorial](https://codeforces.com/blog/entry/68748)
+
+<Spoiler title="Hint">
+
+We can observe that for each legal proposal, reversing the direction of its
+edges creates another legal proposal.
+
+So the sum of costs will be $m\cdot\frac{\texttt{DAGCount}}2$,
+
+where $\texttt{DAGCount}$ represents the number of directed acyclic graphs (DAG)
+generated on the original graph.
+
+</Spoiler>
+
+<Spoiler title="Explanation">
+
+Let $\texttt{dp}[S]$ be the number of ways of construct a directed acyclic
+graphs (DAG) with vertex set $S$ , and $\texttt{indepset}[S]$ represents if $S$
+is an independent set.
+
+Since we're constructing a DAG, we can transfer the state by adding new points.
+Adding 0-out-degree points (or 0-in-degree points is OK) will be easier.
+
+Loop all non-empty subsets of S as points to be added, points are 0-out-degree
+if and only if they don't have edges with each other in this subset. That is,
+the subset is an independent set.
+
+Also, we need to use the principle of inclusion and exclusion.
+
+$$
+dp[S]=\sum_{P \subset S,P\not=\emptyset,\texttt{indepset}[P]=\texttt{true}}(−1)^{|P|+1}dp[S / P]
+$$

```

Comment:
Why is this true? Is it possible to prove this without invoking chromatic polynomials? (https://math.stackexchange.com/questions/2231526/prove-that-number-of-acyclic-orientations-of-a-graph-is-the-chromatic-polynomial)

Reply:
I am trying to write a (informal) proof, but I encountered a difficulty, maybe it'll be finished in a few days

Reply:
might be finished